src/stirling/utils/detect_application.cc

```diff
@@ -49,6 +49,17 @@ Application DetectApplication(const std::filesystem::path& exe) {
   return Application::kUnknown;
 }
 
+std::string GetNodeApplicationFilename(std::string_view cmdline) {
```

Comment:
You could use return value of `std::optional<std::string>` to be more explicit about whether a result was found or not.

Reply:
Updated the return value of this in https://github.com/pixie-io/pixie/pull/1971/commits/92f79bac649483a2caec9dddc4b0dcb003e16e1c.